js/content/site-specific/ddg.js

```diff
@@ -17,20 +17,35 @@
 			const searchResults = document.querySelectorAll('.result');
 			for (const result of searchResults) {
 				const link = result.querySelector('a');
-				if (link) {
-					for (const blockRule of blockList) {
+				if (!link) {
+					break;
+				}
+				
+				for (const blockRule of blockList) {
+					if (!blockRule.type) {
+						break;
+					}
+
+					if (blockRule.type === 'regex') {
 						const regexp = new RegExp(blockRule);
 						const isMatch = regexp.test(link);
 						if (isMatch) {
 							result.classList.add('searchCleanUpFilter');
 							blocked++;
 							break;
 						}
+					} else if (blockRule.type === 'text') {
+						const isMatch = link.innerText.toLowerCase().includes(blockRule.value);
```

Comment:
Atm I only check the innerText, do we want to check anything else? like the url?

Reply:
I think the URL is the most important thing. I might want to say "google" or "w3schools" and it should not depend on the title of the page.
Also I see you are checking against a lowercased linktext while the blockrule doesn't seem to be lowercased.

Reply:
changed to match for `href` and lowercased the value.